### DISCUSSION TITLE
Support specifying WireGuard peer endpoint IP version (IPv4/IPv6)

### DISCUSSION THREAD

**Discussion by EndPositive** [mullvad/mullvadvpn-app#9126] [Feature requests & Ideas]
### I have checked if others have suggested this already

- [X] I have checked this issue tracker to see if others have reported similar issues.

### Feature description

In the iOS app, there's currently no way to specify the wanted IP version for connecting to Mullvad WireGuard servers. I'm in a restricted environment where I only want to use IPv6 WireGuard endpoint. On Linux, it is already possible to select the IP version (see screenshot). On iOS, there's no such option.

![image](https://github.com/user-attachments/assets/4df84c45-e3e3-43e6-b494-ce8403c0de68)

### Alternative solutions

Import IPv6 WireGuard configurations from Mullvad configuration generator and use WireGuard app instead of Mullvad app.

### Type of feature

- [ ] Better privacy/anonymity
- [X] Better at circumventing censorship
- [ ] Easier to use
- [ ] Other

### Operating System

- [ ] Android
- [X] iOS
- [ ] Windows
- [ ] macOS
- [ ] Linux

**Comment by buggmagnet** [comment]
Hi @EndPositive !

Thank you for your support. We have discussed this with the team, and we definitely want to add the feature on iOS.
Whilst I cannot talk about a timeline for when we will have this, I can say with confidence that we will work on it in the future.

In the meantime, if you have trouble connecting to our relays, I would suggest contacting our support team, either directly from within the app, with the "report a problem" feature,  or by sending us an email at [support@mullvadvpn.net](mailto:support@mullvadvpn.net)

**Comment by EndPositive** [comment]
Note that Mullvad on iOS doesn't support IPv6 at all (also not automatic). WireGuard IPv6 is currently not blocked by a certain eastern firewall, so adding it would be super helpful @hulthe

**Comment by artiwl** [comment]
Hello @buggmagnet ,
May I ask if there are any new developments regarding this?
